Issue: 
	kernel panic at abe_block_copy with short tones + capture use cases.

Fix: 
	cancel the workqueue before scheduling it if it is pending
	since the same workqueue has been used for both playback 
	and capture scenarios.

Index: kernel/sound/soc/omap/omap-mcpdm.c
===================================================================
--- kernel.orig/sound/soc/omap/omap-mcpdm.c	2011-06-23 20:25:31.772467858 +0530
+++ kernel/sound/soc/omap/omap-mcpdm.c	2011-06-23 20:25:33.773723586 +0530
@@ -713,7 +713,8 @@
 	mutex_lock(&mcpdm->mutex);
 
 	/* make sure we stop any pre-existing shutdown */
-	cancel_delayed_work(&mcpdm->delayed_abe_work);
+	if (delayed_work_pending(&mcpdm->delayed_abe_work))
+		cancel_delayed_work_sync(&mcpdm->delayed_abe_work); 
 
 	if (!dai->active && mcpdm->free) {
 		ret = omap_mcpdm_request(mcpdm);
@@ -749,6 +750,8 @@
 		mcpdm->ul_active--;
 
 	if (!dai->active) {
+		if (delayed_work_pending(&mcpdm->delayed_abe_work))
+			cancel_delayed_work_sync(&mcpdm->delayed_abe_work); 
 		wake_lock_timeout(&mcpdm->wake_lock, 2 * HZ);
 		schedule_delayed_work(&mcpdm->delayed_abe_work,
 				      msecs_to_jiffies(1000)); /* TODO: pdata ? */
